server/controllers/user/getCampaignById.ts

```diff
@@ -0,0 +1,19 @@
+import { Request, Response, NextFunction } from 'express';
+import { Campaign } from '../../database/models';
+import { CustomedError } from '../../utils';
+import { paramsSchema } from '../../utils/validation';
+
+const getCampaignById = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const result = await paramsSchema.validateAsync(req.params);
+    const { id } = result;
```

Comment:
Why not destructure the id immediately? instead of getting the result and then destructuring